packages/function-tree/src/index.js

```diff
@@ -191,6 +191,11 @@ export class FunctionTree extends EventEmitter {
     this.runTree.on = this.on.bind(this)
     this.runTree.once = this.once.bind(this)
     this.runTree.off = this.removeListener.bind(this)
+
+    // When not extended
+    if (!returnInstance) {
```

Comment:
@christianalfoni have you tested it? looks dangerously for me. Aren't it loose context in this case?